cogent/src/Cogent/DataLayout/Desugar.hs

```diff
@@ -1,27 +1,73 @@
-module Cogent.DataLayout.Desugar where
+module Cogent.DataLayout.Desugar
+ ( desugarAbstractTypeSigil
+ , desugarSigil
+ , desugarDataLayout -- Exposed only for testing
+ ) where
+
 import Data.Map (Map)
 import Data.Map as M
+import Data.Traversable (mapAccumL)
+
+import Text.Parsec.Pos (SourcePos, newPos)
 
-import Cogent.Compiler            ( __fixme
-                                  , __impossible
-                                  )
-import Cogent.Common.Syntax       ( DataLayoutName
-                                  , Size
-                                  )
-import Cogent.Common.Types        ( Sigil(Unboxed, Boxed))
-import Cogent.DataLayout.Surface  ( DataLayoutExpr
-                                  , DataLayoutSize
-                                  , RepSize(Bytes, Bits, Add)
-                                  , RepExpr(RepRef, Prim, Offset, Record, Variant)
-                                  )
+import Cogent.Compiler              ( __fixme
+                                    , __impossible
+                                    )
+import Cogent.Common.Syntax         ( DataLayoutName
+                                    , Size
+                                    , TagName
+                                    , FieldName
+                                    )
+import Cogent.Common.Types          ( Sigil(Unboxed, Boxed))
+import Cogent.DataLayout.Surface    ( DataLayoutExpr
+                                    , DataLayoutSize
+                                    , RepSize(Bytes, Bits, Add)
+                                    , RepExpr(RepRef, Prim, Offset, Record, Variant)
+                                    )
+import Cogent.DataLayout.TypeCheck  (desugarSize)
 import Cogent.DataLayout.Core
+import Cogent.Core                  ( Type (TUnit, TPrim, TSum, TRecord, TCon, TVar, TVarBang, TFun, TString, TArray, TProduct)
+                                    )
+
+{- * DESUGARING 'Sigil's -}
+
+-- | After normalisation, 'TCon _ _ _' values only represent primitive and abstract types.
```

Comment:
After desugaring, instead of nomalisation.

Reply:
@mdimeglio  Oh, if you mean the WH-normalisastion in the surface typechecker, instead of A-nomalisation, then you're right.